### PR TITLE
simplified code

### PR DESCRIPTION
**Pull request by byronwong**
I have simplified the code now that I have a better understanding of what is happening. Please let me know if there are any places I can improve, thanks for all the help!

### CODE REVIEW

File: src/js/index.js [hunk 1]
```diff
@@ -37,91 +37,70 @@ $(document).ready(function () {
...
-  function sortCountry(x, y) {
-    if (x > y) {
-      return 1;
-    }
-    if (x < y) {
-      return -1;
+  function processJobData(data) {
+
+    // compare Fn
+    function sortCity(jobA, jobB) {
```
Comment (panosvoudouris) [inline src/js/index.js]: Is this necessary? In my opinion it makes the code less readable vs having it outside processJobData() and it doesn't add anything.

File: src/js/index.js [hunk 2]
```diff
@@ -37,91 +37,70 @@ $(document).ready(function () {
...
-      return -1;
-    }
-    return 0;
+    // sort by city
+    Object.keys(jobs).map(function (currentItem) {
+      jobs[currentItem].sort(sortCity);
+    });
+
+    // combine country arrays
+    var conbinedJobs = Object.keys(jobs).sort().reduce(function (accumulator, currentItem) {
```
Comment (panosvoudouris) [inline src/js/index.js]: co`m`bined, not co`n`bined

File: src/js/index.js [hunk 3]
```diff
@@ -37,91 +37,70 @@ $(document).ready(function () {
...
+      return accumulator.concat(jobs[currentItem]);
+    }, []);
+
+    return conbinedJobs;
```
Comment (panosvoudouris) [inline src/js/index.js]: co`m`bined, not co`n`bined

File: src/js/index.js [hunk 4]
```diff
@@ -37,91 +37,70 @@ $(document).ready(function () {
...
-              .map(tpl)
-              .join('')
-          );
-        })
-        .catch(function (error) {
-          console.log(error);
-        });
-    })();
+      .catch(function (error) {
+        console.log(error);
```
Comment (panosvoudouris) [inline src/js/index.js]: Is there a plan to put an error message or something else for the user to see if this happens? Right now all they see is the career page with just "...still interested?" but nothing else.

File: src/js/index.js [hunk 5]
```diff
@@ -37,91 +37,70 @@ $(document).ready(function () {
...
-  // Sort By Country
-  function sortCountry(x, y) {
-    if (x > y) {
-      return 1;
-    }
-    if (x < y) {
-      return -1;
+  function processJobData(data) {
+
+    // compare Fn
```
Comment (iamtmrobinson) [inline src/js/index.js]: DEATH TO COMMENTS!

File: src/js/index.js (hunk 1 above)
Comment (byronwong) [inline src/js/index.js]: Thanks, that's a good point especially when it does not have any other use. I'll make the changes

File: src/js/index.js (hunk 2 above)
Comment (byronwong) [inline src/js/index.js]: Thanks, I've probably copied and pasted that throughout. Will amend.

File: src/js/index.js (hunk 4 above)
Comment (byronwong) [inline src/js/index.js]: Yeah, I'm checking with the Recruitment guys to see which positions are going to be open on a permanent basis, and just have the html there as default.

File: src/js/index.js (hunk 5 above)
Comment (byronwong) [inline src/js/index.js]: Sure, I'll remove them, as they don't really add anything

**Comment by shahzainb** [comment]
I'm refactoring the code right now, so this will be redundant. Will go through this with you Byron when I see you.

**Pull request closed without merging**